src/InfiniteIterableValue.php

```diff
@@ -459,6 +511,24 @@ public function last()
         return $value;
     }
 
+    /**
+     * @phpstan-return InfiniteIterableValue<int, TKey>
+     */
+    public function keys(): InfiniteIterableValue
+    {
+        return self::fromStack($this->stack->push(
+            /**
+             * @phpstan-param iterable<TKey, TValue> $iterable
+             * @phpstan-return iterable<TKey, TValue>
```

Comment:
Wouldn't be like `@phpstan-return iterable<int, TKey>`?

Reply:
:+1: